### PR TITLE
rbd: support cloning an image from a non-primary snapshot

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

File: src/test/librbd/image/test_mock_CloneRequest.cc [hunk 1]
```diff
@@ -0,0 +1,763 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (trociny) [inline src/test/librbd/image/test_mock_CloneRequest.cc]: nit: `mode:C++`

**Comment by trociny** [comment]
@dillaman need to remove `expect_failure "is non-primary" clone_image ...` test from qa/workunits/rbd/rbd_mirror.sh.

**Comment by dillaman** [comment]
@trociny thanks -- repushed

**Comment by trociny** [comment]
@ceph-jenkins retest this please

**Pull request merged**